src/main/java/com/github/dregheap/service/DefaultPostService.java

```diff
@@ -0,0 +1,54 @@
+package com.github.dregheap.service;
+
+import com.github.dregheap.model.Post;
+import com.github.dregheap.repository.PostRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class DefaultPostService implements PostService {
+
+    private final PostRepository postRepository;
+
+    @Override
+    public List<Post> findAll() {
+        return postRepository.findAll();
+    }
+
+    @Override
+    public Page<Post> findAll(Pageable pageable) {
+        return postRepository.findAll(pageable);
+    }
+
+    @Override
+    public Post findById(String id) {
+        return postRepository.findOne(id);
+    }
+
+    @Override
+    public Post create(Post post) {
+        post.setId(null);
```

Comment:
For prevent updating   of object with the same id as in request body. This method is only for creating new objects in db.